src/main.js

```diff
@@ -1,101 +1,96 @@
-mirroredMain.add(lineMirroredMain);
-mirroredMain.position.set(0, -mainY, 10);
-scene.add(mirroredMain);
-
-let subLocationX = 0;
-const subDimensions = {depth: 1, width: 1, height: 1};
-const subGeometry = new THREE.BoxGeometry(subDimensions.width, subDimensions.height, subDimensions.depth);
-const subMaterial = new THREE.MeshBasicMaterial({color: 0x0000ff});
-const sub = new THREE.Mesh(subGeometry, subMaterial);
-const edgesSub = new THREE.EdgesGeometry(geometry);
-const lineSub = new THREE.LineSegments(edgesSub, new THREE.LineBasicMaterial({ color: 0x000000 }));
-sub.add(lineSub);
-sub.position.set(-subDimensions.depth/2, 0, subDimensions.height/2);
-scene.add(sub);
+let controls;
+setupControls();
+addAxes();
+const labelRenderer = addAxesLabels();
+
+// PAGE ELEMENTS
+const subConfigCheckbox = document.getElementById("subConfigCheckbox");
+const arraySpanInput = document.getElementById("as");
+const arrayBottomHeightInput = document.getElementById("abz");
+const subDistanceFromCenterInput = document.getElementById("sy");
+const subDepthInput = document.getElementById("sx");
+const audienceDepthFirstRowInput = document.getElementById("axf");
+const audienceDepthLastRowInput = document.getElementById("axl");
+const distanceReferencedFromBelowArrayCheckbox = document.getElementById("xoff");
+const arrayDepthInput = document.getElementById("ad");
+const audienceSeatedRadio = document.getElementById("seated-radio");
+const audienceStandingRadio = document.getElementById("standing-radio");
+const metersRadio = document.getElementById("meters-radio");
+const feetRadio = document.getElementById("feet-radio");
+
+// OBJECTS
+let mainDimensions = new Dimensions(1, 1, 1);
+setMainDimensionHeight(arraySpanInput.value);
+setMainDimensionDepth(arrayDepthInput.value);
 
+let mainLocation = setMainLocation();
+let main = createCube(mainLocation, 0xff0000, mainDimensions);
+setMainZFromBottom(arrayBottomHeightInput.value);
+setMainYFromSub(subDistanceFromCenterInput.value);
 
+let mainMirrorLocation = new Location(main.position.x, -main.position.y, main.position.z);
+let mainMirror = createCube(mainMirrorLocation, 0xff0000, mainDimensions);
+
+const subDimensions = new Dimensions(1, 2, 1);
+let subLocation = setSubLocation();
+let sub = createCube(subLocation, 0x0000ff, subDimensions);
+let subMirror;
+setSubLocationY(subConfigCheckbox.checked, subDistanceFromCenterInput.value);
+
+addSubMirror();
+
+let audienceDimensions = new Dimensions(0.1, main.position.y * 4, main.position.y * 4);
+let audienceLocation = new Location(audienceDimensions.depth / 2 + 5, 0, 1.2);
+let audience = createCube(audienceLocation, 0x00ff00, audienceDimensions, 0.15);
+updateAudience(audienceDepthFirstRowInput.value, audienceDepthLastRowInput.value, subDistanceFromCenterInput.value, distanceReferencedFromBelowArrayCheckbox.checked,
+    audienceSeatedRadio.checked, metersRadio.checked);
+
+fitCameraToSelection();
```

Comment:
This could be the end of a "setup function", that could initialize globals, three.js and the main three objects.

Reply:
Hmmm, I'm having trouble picturing it. I guess the result would be all global variables in a function called setup(), which sounds fine, but I'm not clear on what the benefits would be and what the effects of it would be.